aces/Loris
# [Core] Add modules table and make modules responsible for maintaining LORIS menu

There was no previously centralized list of modules in the LORIS database, but the module descriptor was lazily loaded when trying to access a module. As a result features such as the menu required a centralized configuration (at one point in config.xml, now in the `LorisMenu` table), modules could not be deactivated (other than by deleting them) and the menus table had to be manually maintained (by our release patches, with no front end to maintain it) including ensuring the permissions were synchronized with the code.

This adds a `modules` table which lists modules that are available (and still needs to be maintained by our release patches currently.) This allows us to add a `getMenuItems` function to the \Module class, which modules can use the register which menu items should be added to LORIS menu and can be managed by the module itself without SQL access. It also ensures that the menu items are directly tied to the user's permissions by using the same `hasAccess` function that the router uses to validate, rather than the approximation of the `LorisMenuPermissions` table.

Modules must now implement a `getLongName` function to return the human friendly name for the module to be used by default in menus (but can also be used other places that a human readable version of the module name is needed in the future.)

## Testing

1. Run the patch `SQL/New_patches/2019-12-05-AddModuleTable.sql`
2. Ensure that the menu links should all go to the same places
3. You can also change the Active flag for a module or two and they should be removed from the menu.. but since the Modules table should have every module in it in the patch provided, and there's no way to currently change the active flag, that's not something that users can do right now.
